src/WingetCreateCLI/Commands/NewCommand.cs

```diff
@@ -237,10 +239,130 @@ private static void PromptRequiredProperties<T>(T manifest, VersionManifest vers
                         continue;
                     }
 
-                    var currentValue = property.GetValue(manifest);
-                    var result = PromptProperty(manifest, currentValue, property.Name);
-                    property.SetValue(manifest, result);
-                    Logger.Trace($"Property [{property.Name}] set to the value [{result}]");
+                    PromptHelper.PromptAndSetPropertyValue(manifest, property.Name);
+                    Logger.Trace($"Property [{property.Name}] set to the value [{property.GetValue(manifest)}]");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Displays all installers from an Installer manifest as a selection menu.
+        /// </summary>
+        private static void DisplayInstallersAsMenuSelection(InstallerManifest installerManifest)
+        {
+            Console.Clear();
+
+            while (true)
+            {
+                List<string> selectionList = GenerateInstallerSelectionList(installerManifest.Installers);
+                var selectedItem = Prompt.Select(Resources.SelectInstallerToEdit_Message, selectionList);
+
+                if (selectedItem == Resources.None_MenuItem)
+                {
+                    break;
+                }
+                else if (selectedItem == Resources.AllInstallers_MenuItem)
+                {
+                    Installer installerCopy = new Installer();
+                    PromptHelper.DisplayPropertiesAsMenuSelection(installerCopy, Resources.None_MenuItem);
+                    ApplyChangesToIndividualInstallers(installerCopy, installerManifest.Installers);
+                }
+                else if (selectedItem == Resources.DisplayPreview_MenuItem)
+                {
+                    Console.Clear();
+                    Console.WriteLine();
+                    Logger.InfoLocalized(nameof(Resources.DisplayPreviewOfSelectedInstaller_Message));
+                    var serializer = Serialization.CreateSerializer();
+                    string installerString = serializer.Serialize(installerManifest);
+                    Console.WriteLine(installerString);
+                    Console.WriteLine();
+                }
+                else
+                {
+                    Installer selectedInstaller = MatchMenuSelectionToInstaller(selectedItem, installerManifest);
+                    PromptHelper.DisplayPropertiesAsMenuSelection(selectedInstaller, Resources.None_MenuItem);
+                }
+            }
+        }
+
+        private static Installer MatchMenuSelectionToInstaller(string selectedItem, InstallerManifest installerManifest)
+        {
+            string[] selection = selectedItem.Split('|');
+            var matchedInstallers = installerManifest.Installers.Where(
+                item => item.Architecture == (InstallerArchitecture)Enum.Parse(typeof(InstallerArchitecture), selection[0]) &&

```

Comment:
>Enum.Parse( [](http://example.com/codeflow?start=68&length=11)

Can use Enum.Parse<InstallerArchitecture(selection[0])

Reply:
changed to suggestion